### ISSUE TITLE
Although per_device_train_batch_size*gradient_accumulation_steps is consistent, the loss decline curve is inconsistent

### ISSUE THREAD

**Issue by Linjyan00** [haotian-liu/LLaVA#1407]
For the training stage of llava1.5 (LLaVA/scripts/v1_5/finetune_lora.sh), I tried the following two different examples and found something wrong (2*A100)

--per_device_train_batch_size 4 \
--gradient_accumulation_steps 16 \

--per_device_train_batch_size 1 \
--gradient_accumulation_steps 64 \

In these two Settings, their first-iteration loss is not the same, but by definition, they should be. Why?
